core/src/admin_rpc_post_init.rs

```diff
@@ -8,19 +8,79 @@ use {
     solana_quic_definitions::NotifyKeyUpdate,
     solana_runtime::bank_forks::BankForks,
     std::{
-        collections::HashSet,
+        collections::{HashMap, HashSet},
         net::UdpSocket,
         sync::{Arc, RwLock},
     },
 };
 
+/// Key updaters:
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub enum KeyUpdaterType {
+    /// TPU key updater
+    Tpu,
+    /// TPU forwards key updater
+    TpuForwards,
+    /// TPU vote key updater
+    TpuVote,
+    /// Forward key updater
+    Forward,
+    /// For the RPC service
+    RpcService,
+    /// For the connection cache
+    ConnectionCache,
+}
+
+/// Responsible for managing the updaters for identity key change
+#[derive(Default)]
+pub struct KeyUpdaters {
+    updaters: HashMap<KeyUpdaterType, Arc<dyn NotifyKeyUpdate + Sync + Send>>,
+}
+
+impl KeyUpdaters {
+    /// Add a new key updater to the list
+    pub fn add(
+        &mut self,
+        updater_type: KeyUpdaterType,
+        updater: Arc<dyn NotifyKeyUpdate + Sync + Send>,
+    ) {
+        self.updaters.insert(updater_type, updater);
+    }
+
+    /// Remove a key updater by its key
+    pub fn remove(&mut self, updater_type: &KeyUpdaterType) {
```

Comment:
I don't understand how this structure and specifically `remove` are going to be used when subscribe/unsubscribe from the Vortexor. If there is a follow up PR on that, would help.

Reply:
The idea is native Tpu can be turned on and off. When a vortexor receiver is enabled, the native Tpu is turned off. If we detected vortexor having problems, we can switch back to the native Tpu. During this we need to unsubscribe and subscribe the key updater callback

Reply:
For a preview -- see https://github.com/anza-xyz/agave/blob/2e965d9da02b9086a8ff6a87a4dee6bfe6684f6a/core/src/tpu_switch.rs#L241

Reply:
Will review and add -- tracked by https://github.com/anza-xyz/agave/issues/6256.